### ISSUE TITLE
Cannot hire workers after relocating Level 5 buildings

### ISSUE THREAD

**Issue by NatoriGitHub** [ldtteam/minecolonies#9372]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [ ] Single Player
- [X] Multi Player

### Minecraft Version

1.20.1

### MineColonies Version

1.20.1-1.1.141-BETA

### Structurize Version

1.20.1-1.0.631-BETA.jar

### Related Mods and their Versions

multipiston-1.20-1.2.31-ALPHA.jar
blockui-1.20.1-0.0.94-ALPHA.jar
domum_ornamentum-1.20-1.0.129-BETA-universal.jar

### Current Behavior

When deconstructing a level 5 build and rebuilding, once the building is complete, I am unable to hire the worker. The hire button does not appear. This happens to multiple buildings, not just one type. My work-around is to trash the until and make a new one and starting from level 1. 

### Expected Behavior

I expect that when I am rebuild a level 5 building, I should be able to hire the worker that was already working.

### Reproduction Steps

1. deconstruct level 5 building
2. rebuild using that same building unit
3. attempt to hire worker

### Logs

<script src="https://gist.github.com/NatoriGitHub/811e0ae79331d960bf64b86247892898.js"></script>

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Corbi84** [comment]
Had the same happen in the latest version of Colony Venture (2.10). Relocated a Couriers Hut and couldn't re-hire anyone after rebuilding nor after repairing. Worked again normally after upgrading the hut from Lvl2 to Lvl3.

Versions:
Minecraft 1.19.2
Forge 43.2.13
minecolonies-1.19.2-1.1.151-BETA.jar
structurize-1.19.2-1.0.630-RELEASE.jar
-Singleplayer

**Comment by na033681** [comment]
I have the same issue. Only solution I have found was going to creative and overwriting a rebuild when its completed with a new building of the same type. That breaks the original bugged hut block and replaces a new one that can hire.

Versions:
Minecraft 1.19.2
Forge 43.2.21
MineColonies 1.19.2-1.1.151-Beta
Structurize 1.19.2-1.0.630-Release
Domum Ornamentum 1.19-1.0.130-beta-universal
-Singleplayer and LAN

**Comment by Estyria** [comment]
We are having a lot of reports about this on the discord for both supported versions. After repositioning, workers cannot be hired at the huts.

**Comment by fernanfs** [comment]
I noticed some interesting things:

I placed the hut block (after is was previously built as a level 5 building) and tasked my builder to build the hut. Once it was successfully completed, the builder stated that it has *repaired* the building, not "finished building". This seemed quite odd.
I tried to reproduce and noticed, that there was no Deconstruct button, even though the building had been fully built. Instead I was able to pickup the hut without prior deconstruction.

Thus my guess would be, that there is some inconsistent state somewhere about the state of the building. Does anybody know where the state is actually stored? I compared the NBT data of a working building with a defect one and coudn't see that much of a difference.
